src/auth.rs

```diff
@@ -108,31 +101,22 @@ impl Authenticator {
                                                       String,
                                                       u32);
 
-        let rnonce_b64 = match rnonce_opt {
-            Some(val) => val,
-            None => return Err(ResponseError(String::from("Invalid rnonce returned"))),
-        };
+        let rnonce_b64 = rnonce_opt
+            .ok_or(ResponseError(String::from("Invalid rnonce returned")))?;
```

Comment:
Good call, I forgot about that one. I remember clippy suggests that; I wonder if it's possible to set up clippy as a CI tool to check PRs?